src/macro.c

```diff
@@ -376,7 +376,19 @@ void Macro::expand(OutBuffer *buf, size_t start, size_t *pend,
                     continue;
                 }
 
+                bool macroWasUndefined = false;
                 Macro *m = search(name, namelen);
+
+                if (!m)
+                {
+                    static const char undef[] = "DDOC_UNDEFINED_MACRO";
+                    m = search((const utf8_t *)undef, sizeof(undef) - 1);
```

Comment:
Copy any of the similar code in `doc.c` eg https://github.com/andralex/dmd/blob/DDOC_UNDEFINED_MACRO/src/doc.c#L2229